pkg/apis/app/v1alpha1/application_types.go

```diff
@@ -100,6 +101,34 @@ type InfoItem struct {
 
 	// Value is human readable content.
 	Value string `json:"value,omitempty"`
+
```

Comment:
You are implementing the golang version of a union. You need to add a type specified to indicate whether Value or ValueFrom is set e.g. 
```go
type InfoItemType string

const (
  Value InfoItemType = "value"
  Reference = "reference"
)
```
You may also want to make Value a pointer. Only one should be set, and it should correspond to the indicated type field.

Reply:
Understood your comment about union in Go. Such type wouldn't be consistent with what is currently in the core API though. I'm drawing from the following examples:

https://github.com/kubernetes/api/blob/184e700b32b7f1b532b9fce8dd8c1f412d297c4b/core/v1/types.go#L1663